lib/address.py

```diff
@@ -243,30 +243,43 @@ def from_string(self, string):
                 script.extend(Script.push_data(binascii.unhexlify(word)))
         return ScriptOutput(bytes(script))
 
-    def to_ui_string(self, hex_only = False):
-        '''Convert to user-readable OP-codes (plus text), eg OP_RETURN (12) Hello there!
-           Or, to a hexadecimal string if that fails.
-           Note that this function is the inverse of from_string() only if called with hex_only = True!'''
-        if self.script and not hex_only:
-            try:
-                ret = ''
-                ops = Script.get_ops(self.script)
-                for op in ops:
-                    if ret: ret += ", "
-                    def lookup(x):
-                        return OpCodes.reverseLookup.get(x, ('('+str(x)+')'))
-                    if isinstance(op, tuple):
-                        opData = op[1].decode(errors='replace') if isinstance(op[1], bytes) else str(op[1])
-                        ret += lookup(op[0]) + " " + opData
-                    elif isinstance(op, int):
-                        ret += lookup(op)
-                    else:
-                        ret += '[' + (op.hex() if isinstance(op, bytes) else str(op)) + ']'
-                return ret
-            except ScriptError:
-                # Truncated script -- so just default to normal 'hex' encoding below.
-                pass      
-        return self.script.hex()
+    def to_ui_string(self,):
+        '''Convert to user-readable OP-codes (plus pushdata as text if possible)
+        eg OP_RETURN (12) "Hello there!"
+        '''
+        try:
+            ops = Script.get_ops(self.script)
+            parts = []
+            for op in ops:
+                def lookup(x):
+                    return OpCodes.reverseLookup.get(x, ('('+str(x)+')'))
+                if isinstance(op, tuple):
+                    op, data = op
+                    if data is None:
+                        data = b''
+
+                    # Attempt to make a friendly string, or fail to hex
+                    try:
+                        astext = data.decode('utf8')
+
+                        friendlystring = repr(astext)
+
+                        # if too many escaped characters, it's too ugly!
+                        if friendlystring.count('\\')*3 > len(astext):
+                            friendlystring = None
+                    except:
+                        friendlystring = None
+
+                    if not friendlystring:
+                        friendlystring = data.hex()
+
+                    parts.append(lookup(op) + " " + friendlystring)
+                else: # isinstance(op, int):
+                    parts.append(lookup(op))
+            return ', '.join(parts)
+        except ScriptError:
```

Comment:
What is this error?  Did you get any insight into it in your work on this?

Reply:
Yeah, actually this has to do with truncated scripts as I mentioned in a recent (fixed) Issue https://github.com/fyookball/electrum/issues/829. Under my PR would look like:

![screenshot from 2018-09-18 13-55-29](https://user-images.githubusercontent.com/36528214/45716017-8d748d80-bb4a-11e8-9f80-1af4a1016c85.png)